src/Nancy.Hosting.Self/NancyHost.cs

```diff
@@ -426,24 +426,31 @@ private void GotCallback(IAsyncResult ar)
 
         private void Process(HttpListenerContext ctx)
         {
-            try
+            Action<Exception> onError = (e) =>
             {
-                var nancyRequest = this.ConvertRequestToNancyRequest(ctx.Request);
-                using (var nancyContext = this.engine.HandleRequest(nancyRequest))
+                this.configuration.UnhandledExceptionCallback.Invoke(e);
+            };
+
+            Action<NancyContext> onComplete = (nancyContext) =>
+            {
+                try
                 {
-                    try
-                    {
-                        ConvertNancyResponseToResponse(nancyContext.Response, ctx.Response);
-                    }
-                    catch (Exception e)
-                    {
-                        this.configuration.UnhandledExceptionCallback.Invoke(e);
-                    }
+                    ConvertNancyResponseToResponse(nancyContext.Response, ctx.Response);
                 }
+                catch (Exception e)
+                {
+                    onError(e);
+                }
+            };
+
+            try
+            {
+                var nancyRequest = this.ConvertRequestToNancyRequest(ctx.Request);
+                this.engine.HandleRequest(nancyRequest, onComplete, onError);

```

Comment:
Where is this overload defined? I think I must be blind, cause I just can't find it 😕

Reply:
I’m on my phone so I can’t be sure, but I believe this is it. Note that this is on the 1.x branch since I was working against stable.

https://github.com/NancyFx/Nancy/blob/b826aaf6c00b51804c0de9542c1f7c13a3ef2604/src/Nancy/NancyEngineExtensions.cs#L78